src/models/ingestion/ingestionHandler.ts

```diff
@@ -0,0 +1,213 @@
+      metricsCollector,
+      stateManager,
+      generateFeatureId: false,
+    });
+
+    this.logger.info({ msg: 'shapefile chunk reader initialized', maxVerticesPerChunk: this.maxVerticesPerChunk });
+    return reader;
+  }
+
+  private setupChunkProcessor(job: IJobResponse<IngestionJobParams, unknown>): ChunkProcessor {
+    return {
+      process: async (chunk: ShapefileChunk): Promise<void> => {
+        this.logger.info({
+          msg: 'processing chunk',
+          chunkId: chunk.id,
+          verticesCount: chunk.verticesCount,
+          chunkSize: chunk.features.length,
+          skippedFeaturesCount: chunk.skippedFeatures.length,
+        });
+        // const { skippedFeatures } = chunk; TODO: handle skipped features(Report them)
+
+        const validFeatureCollection = this.parseChunk(chunk, job);
+
+        await this.handleChunk(job, validFeatureCollection);
+        this.logger.info({ msg: 'chunk processed', chunkId: chunk.id, featuresCount: validFeatureCollection.features.length });
+      },
+    };
+  }
+
+  private parseChunk(chunk: ShapefileChunk, job: IJobResponse<IngestionJobParams, unknown>): PolygonPartsFeatureCollection {
+    const shpFeatures = shpFeatureSchema.array().safeParse(chunk.features);
+    if (!shpFeatures.success) {
+      this.logger.error({ msg: 'error validating features in chunk', chunkId: chunk.id, errors: shpFeatures.error.errors });
+      throw new ZodError(shpFeatures.error.issues); //TODO: report which features are invalid instead of throwing an error
+    }
+
+    const featureResolutions: FeatureResolutions = {
+      resolutionMeter: calculateResMeterFromDegree(job.parameters.ingestionResolution),
+      resolutionDegree: job.parameters.ingestionResolution,
+    };
+
+    const mappedFeatures = shpFeatures.data.map((feature) => {
+      const mappedProperties = this.mapShpPropertiesSchemaToPartProperties(feature.properties, featureResolutions);
+      return {
+        ...feature,
+        properties: mappedProperties,
+      };
+    });
+    return { type: 'FeatureCollection', features: mappedFeatures };
+  }
+
+  private validateShapefilesExists(shapefileRelativePath: string): void {
+    const shapefileFullPath = path.join(this.ingestionSourcesDirPath, shapefileRelativePath);
+    const parsed = path.parse(shapefileFullPath);
+    const basePath = path.join(parsed.dir, parsed.name);
+    const missing: string[] = [];
+
+    for (const ext of SHAPEFILE_EXTENSIONS_LIST) {
+      const filePath = basePath + ext;
+      if (!fs.existsSync(filePath)) {
+        missing.push(filePath);
+      }
+    }
+
+    if (missing.length > 0) {
+      this.logger.error({ msg: 'shapefiles are missing required files', basePath, missingFiles: missing });
+      throw new ShapefileNotFoundError(basePath, missing);
+    }
+  }
+
+  private mapShpPropertiesSchemaToPartProperties(shpProps: ShpFeatureProperties, resolutions: FeatureResolutions): PartFeatureProperties {
```

Comment:
this a simple property mapper used once
consider removing or perhaps extending it's responsibility to map shpFeatures in the chunk